### PR TITLE
[storage] Fix index flush on alignment

### PR DESCRIPTION
**Pull request by dentiny**
## Summary

The bug is:
- When alignment reaches boundary and requests to flush, the last flush only writes the active buffer and switch buffer index
- Now we have a complete byte left in the active buffer unflushed, the fix here is to flush twice: one for active buffer, one for the left byte

It's hard to write a proper unit test, since it's hard to trigger such boundary case :(
I checked with the previously failed chaos test replay, and confirmed it works now.

Thanks @zhousun for saving my life :sob:

## Related Issues

Closes https://github.com/Mooncake-Labs/moonlink/issues/1867

## Checklist

- [x] Code builds correctly
- [ ] Tests have been added or updated
- [ ] Documentation updated if necessary
- [x] I have reviewed my own changes

### CODE REVIEW

File: src/moonlink/src/storage/async_bitwriter.rs [hunk 1]
```diff
@@ -175,6 +175,7 @@ impl<W: AsyncWrite + Unpin + Send + Sync, E: Endianness> BitWriter<W, E> {
     /// Pads the current bit queue with zeros until aligned to the next byte boundary.
     /// This should be called before the final flush to ensure the output is byte-aligned.
     /// Returns `true` if a flush is required due to buffer spill.
+    #[must_use]
```
Comment (dentiny) [inline src/moonlink/src/storage/async_bitwriter.rs:178]: A post-merge second thought, if alignment is only involved at close, it's better to handle the `align_and_flush_all` semantics inside of bit writer.
